keps/sig-node/20181106-in-place-update-of-pod-resources.md

```diff
@@ -172,9 +184,24 @@ available memory are more probable to require restarts.
 If more than one resource type with different policies are updated, then
 RestartContainer policy takes precedence over NoRestart policy.
 
-Additionally, if RestartPolicy is 'Never', ResizePolicy should be set to
+Additionally, if RestartPolicy is 'Never', ResizePolicy must be set to
 NoRestart in order to pass validation.
 
+<<[UNRESOLVED]>>
+Should something like RuntimeClass be allowed to say "no in-place restarts are
+allowed", so that users would get a synchronous error if they set this?  Or is
+that overkill?`
+<<[/UNRESOLVED]>>
+
+<<[UNRESOLVED]>>
+`resizePolicy: NoRestart` is not obvious.  To me that reads as "you may not
+restart this, but that's not what it means, right?  Can we phrase itbetter?
+
+Do we need a policy like "RestartNotAllowed" which says the opposite of
+NoRestart - you MAY NOT restart me?  That seems complicated and will leave
+resizes in "limbo", so maybe not.
```

Comment:
These policy values are intended to be user guidance on how resize should be handled by node, they are not guarantees. They may resize CPU + memory, but if resize policy for memory says Restart while CPU is NoRestart, then we won't be calling UpdateContainerResources CRI for CPU at all, we would stop and start the container with new desired CPU and memory values, since memory would require restart anyways. Also, while it's not implemented today, kubelet should have the freedom to restart despite NoRestart policy - e.g if UpdateContainerResources errors out say N times.

So a policy like RestartNotAllowed would render something unenforceable. But RestartOnGrow, RestartOnShrink you suggested on the PR may come in handy.

Reply:
I'm willing to buy that "may not restart" is not useful enough to support.  For now.